lib/hash-tree.js

```diff
@@ -36,7 +38,20 @@ function getFileInfos(fullPath, _visited) {
         return paths;
       }
 
-      return paths.concat(getFileInfos(path.join(realPath, entry), visited));
+      var keys;
+      try {
+        keys = getFileInfos(path.join(realPath, entry), visited);
+      } catch (err) {
+        if (!keysForTreeWarningPrinted) {
```

Comment:
Is this warning actionable?

Reply:
We should only catch the exact error we want to catch, and rethrow the rest.

Reply:
added the rethrowing, and converted the comment to console warning to hopefully be more actionable.